test/testdata/rewriter/chained_sig.rb

```diff
@@ -0,0 +1,182 @@
+
+  interface!
+
+  sig.abstract { void }
+  def foo; end
+end
+
+class DuplicateBlock
+  extend T::Sig
+  include Interface
+
+  sig.override { void }.final { void }
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Cannot add more signature statements after the declaration block
+  def foo; end
+end
+
+class WithoutRuntime
+  extend T::Sig
+  include Interface
+
+  T::Sig::WithoutRuntime.sig.final { void }
+  def baz; end
+
+  T::Sig::WithoutRuntime.sig.params(a: Integer)
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Cannot use `params` outside of a sig block
+                           # ^^^^^^ error: Method `params` does not exist on `NilClass`
+  def bar(a); end
+
+  T::Sig::WithoutRuntime.sig.override { void }.final { void }
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Cannot add more signature statements after the declaration block
+  def foo; end
+end
+
+class MissingBlocks
+  extend T::Sig
+
+  sig
+# ^^^ error: Signature declarations expect a block
+  def foo; end
+
+  sig.final
+# ^^^^^^^^^ error: Signature declarations expect a block
+    # ^^^^^ error: Method `final` does not exist on `NilClass`
+  def bar; end
+end
+
+class InstructionSequenceInSig
+  extend T::Sig
+
+  sig.final { Kernel.puts ""; void }
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Malformed signature: cannot have multiple instructions inside a signature block
+    # ^^^^^ error: Method `final` does not exist on `NilClass`
+                            # ^^^^ error: Method `void` does not exist on `T.class_of(InstructionSequenceInSig)`
+  def foo; end
+end
+
+class DuplicateInvocations
+  extend T::Sig
+  extend T::Helpers
+  include Interface
+
+  abstract!
+
+  sig.overridable { overridable.void }
+                  # ^^^^^^^^^^^ error: Duplicate invocation of `overridable` in signature declaration
+  def bar; end
+
+  sig.override { override.void }
+               # ^^^^^^^^ error: Duplicate invocation of `override` in signature declaration
+  def foo; end
+
+  sig.abstract { abstract.void }
+               # ^^^^^^^^ error: Duplicate invocation of `abstract` in signature declaration
+  def baz; end
+end
+
+class DuplicateButMissingBlock
+  extend T::Sig
+  include Interface
+
+  sig.final { void }.override
+                   # ^^^^^^^^ error: Method `override` does not exist on `NilClass`
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Signature declarations expect a block
+  def foo; end
+# ^^^^^^^ error: Method `DuplicateButMissingBlock#foo` implements an abstract method `Interface#foo` but is not declared with `override.`
+end
+
+class InvalidInvocationAfterBlock
+  extend T::Sig
+
+  sig.final { void }.checked(:never)
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Cannot use `checked` outside of a sig block
+                   # ^^^^^^^ error: Method `checked` does not exist on `NilClass`
+  def foo; end
+end
+
+class ValidDoubleChain
+  extend T::Sig
+  include Interface
+
+  sig.override.final { void }
+  def foo; end
```

Comment:
We also need tests -- ideally for all the cases that use `.final` combined with something else -- that the `sig(:final)` syntax produces the same result.  So here, we would want a separate test that uses `sig(:final).override { void }`.